homeassistant/components/gpslogger/device_tracker.py

```diff
@@ -137,40 +148,53 @@ async def async_added_to_hass(self) -> None:
             self._location = (None, None)
             self._accuracy = None
             self._attributes = {
-                ATTR_ALTITUDE: None,
                 ATTR_ACTIVITY: None,
+                ATTR_ALTITUDE: None,
+                ATTR_BATTERY_CHARGING: None,
                 ATTR_DIRECTION: None,
+                ATTR_LAST_SEEN: None,
                 ATTR_PROVIDER: None,
                 ATTR_SPEED: None,
             }
             self._battery = None
+            self._prv_seen = None
             return
 
         attr = state.attributes
         self._location = (attr.get(ATTR_LATITUDE), attr.get(ATTR_LONGITUDE))
         self._accuracy = attr.get(ATTR_GPS_ACCURACY)
+        last_seen = cast(datetime | None, attr.get(ATTR_LAST_SEEN))
+        self._prv_seen = last_seen
         self._attributes = {
-            ATTR_ALTITUDE: attr.get(ATTR_ALTITUDE),
             ATTR_ACTIVITY: attr.get(ATTR_ACTIVITY),
+            ATTR_ALTITUDE: attr.get(ATTR_ALTITUDE),
+            ATTR_BATTERY_CHARGING: attr.get(ATTR_BATTERY_CHARGING),
             ATTR_DIRECTION: attr.get(ATTR_DIRECTION),
+            ATTR_LAST_SEEN: last_seen,
```

Comment:
Last seen should be a timestamp sensor entity.

Reply:
It is much easier to compare multiple device_tracker entities for a single physical device (e.g., when using multiple presence detection integrations) when last_seen is an attribute, not a separate entity. And there is already precendence for a timestamp like this to be an attribute. E.g., last_seen in (the now defunct) Life360, Google Maps' last_seen, Tile's last_timestamp, and possibly even asuswrt's last_time_reachable. It is crucial for my custom composite integration that many, many people use.

Reply:
FWIW, I understand splitting out miscellaneous data into separate sensors as opposed to the original tendency to keep adding attributes to an existing entity. I have no problem with that in general.

However, in this case, I would argue that this is a perfect example of what an attribute should be used for. It is adding information about the main state (latitude, longitude & accuracy) that makes using that data more accurate. I.e., it describes when the data actually changed (as opposed to the last_updated field, that just indicates when the data was received by HA, which can be, and often is, very different.)